lib/HLSL/DxilValidation.cpp

```diff
@@ -2345,7 +2345,83 @@ static void ValidateDxilOperationCallInProfile(CallInst *CI,
           CI, ValidationRule::SmOpcodeInInvalidFunction,
           {"CreateHandle", "Shader model 6.5 and below"});
     }
+  } break;
+    
+  // this intrinsic is generated when the SV_DispatchThreadID system value is used
+  case DXIL::OpCode::ThreadId: {  
+    // get the node launch type
+    DxilModule &DM = ValCtx.DxilMod;
+    if (!DM.HasDxilEntryProps(F)) {
+      break;
+    }
+
+    DxilEntryProps &entryProps = DM.GetDxilEntryProps(F);
+    DXIL::NodeLaunchType NLT = entryProps.props.Node.LaunchType;
+    if (NLT == DXIL::NodeLaunchType::Thread || NLT == DXIL::NodeLaunchType::Coalescing) {
```

Comment:
I feel this approach is a bit fragile, since it will permit any new launch type if we forget to update the code here.  If we checked for the specific launch types where it's legal instead, that would force us to update the code if necessary in the future, instead of opening up a validation gap.